vgaidarji/ci-matters
# Update project and make it buildable again

### What has been done
- Migrated from jcenter to mavenCentral
- Migrated from Java to Kotlin
- Updated Gradle to 8.4 version 
- Migrated from Fabric.Crashlytics to Firebase
- Removed PMD, Checkstyle configurations as they not needed for Kotlin projects
- Migrated to AndroidX and removed Butterknife in exchange of ViewBinding
- Moved from Coveralls to codecov.io as Coveralls doesn't work well with Kotlin source code
- Updated GitHub Actions CI/CD 
- Updated CircleCI CI/CD
- Updated Bitrise CI/CD

What's left (will be the scope of another PRs)
- [ ] Fix Jenkins
- [ ] Fix Teamcity (and use new Teamcity Pipelines)
- [ ] Configure ktlint

### How to test
- Verify build statuses on all fixed CIs and make sure nothing is wrong in logs (steps are not executed, coverage is not collected, etc.)
- Make sure builds distributed to Firebase
- Make sure code coverage is uploaded to codecov.io
